src/protocols/message_counter/MessageCounterManager.h

```diff
@@ -121,26 +137,30 @@ class MessageCounterSyncMgr : public Messaging::ExchangeDelegate
      */
     struct ReceiveTableEntry
     {
-        System::PacketBufferHandle msgBuf; /**< A handle to the PacketBuffer object holding
-                                                the message data. This is non-null if and only
-                                                if this entry is in use. */
+        ReceiveTableEntry() : peerNodeId(kUndefinedNodeId) {}
+
+        NodeId peerNodeId;                  /**< The peerNodeId of the message. kUndefinedNodeId if is not in use. */
+        PacketHeader packetHeader;          /**< The packet header for the message. */
```

Comment:
This is undoing the removal of the packetHeader from this struct which we did to save memory....  Same for the peerNodeId.  If we have to add the peerAddress in here, ok, though we should figure out why it's really needed by consumers, that can be a followup.  But we shouldn't need the node id or packet header here.

Reply:
According the spec, the reprocessing of the packet should start from `SecureSessionManager::OnMessageReceived`
whose arguments are `PacketHeader`, `PeerAddress` and `msgBuf`. Maybe we can refactor the `SecureSessionManager::OnMessageReceived` function to add another entry to eliminate the requirement of `packetHeader`.

Reply:
I know what the spec says (insofar as it says anything concrete), but the point is the packet header is already in the message.  We don't need to store it separately.